source/AMDSClient.cpp

```diff
@@ -138,96 +142,90 @@ AMDSClient::AMDSClient(QWidget *parent)
-
-	if (blockSize == 0) {
-		if (tcpSocket->bytesAvailable() < (int)sizeof(quint32))
-			return;
-
-		in >> blockSize;
-	}
-
-//	qDebug() << "Received from: " << tcpSocket->peerAddress();
-//	qDebug() << "\n\n\n";
-//	qDebug() << "Block size = " << blockSize;
-//	qDebug() << "Bytes available = " << tcpSocket->bytesAvailable();
-
-	if (tcpSocket->bytesAvailable() < blockSize)
-		return;
-
-	AMDSClientRequest *clientRequest = in.decodeAndInstantiateClientRequestType();
-	in.read(*clientRequest);
-
-	switch (clientRequest->requestType()) {
-	case AMDSClientRequestDefinitions::Introspection: {
-		/// TODO: this need to be reorganized and implemented (printData) in the class of AMDSClientIntrospectionRequest
-		AMDSClientIntrospectionRequest *clientIntrospectionRequest = qobject_cast<AMDSClientIntrospectionRequest*>(clientRequest);
-		if(clientIntrospectionRequest){
-			QList<AMDSBufferGroupInfo> bufferGroupInfos;
-			bufferGroupInfos = clientIntrospectionRequest->bufferGroupInfos();
-
-			if(bufferGroupInfos.count() > 0){
-				const QStandardItemModel* model = qobject_cast<const QStandardItemModel*>(requestType->model());
-				for(int x = 1; x < 7; x++){
-					QStandardItem* item = model->item(x);
-					item->setFlags(Qt::ItemIsSelectable|Qt::ItemIsEnabled);
-					// visually disable by greying out - works only if combobox has been painted already and palette returns the wanted color
-					item->setData(QVariant(), Qt::TextColorRole);
-				}
-			}
+	quint8 requestTypeId = (quint8)requestType->currentIndex();
+	QString bufferName = bufferNameComboBox_->currentText();
+	QDateTime time1 = time1Edit->dateTime();
+	QDateTime time2 = time2Edit->dateTime();
+	QString value1 = count1Edit->text();
+	QString value2 = count2Edit->text();
+	QString continuousSocket = activeContinuousConnection->currentText();
 
-			if(clientIntrospectionRequest->bufferName() == "All"){
-				bufferNameComboBox_->clear();
-				bufferNameComboBox_->addItem("All");
-			}
-			for(int y = 0, ySize = bufferGroupInfos.count(); y < ySize; y++){
-				if(clientIntrospectionRequest->bufferName() == "All")
-					bufferNameComboBox_->addItem(bufferGroupInfos.at(y).name());
-				qDebug() << bufferGroupInfos.at(y).name() << bufferGroupInfos.at(y).description() << bufferGroupInfos.at(y).units() << bufferGroupInfos.at(y).rank() << bufferGroupInfos.at(y).size().toString();
-				for(int x = 0, size = bufferGroupInfos.at(y).axes().count(); x < size; x++){
-					qDebug() << "\tAxis info at " << x << bufferGroupInfos.at(y).axes().at(x).name() << bufferGroupInfos.at(y).axes().at(x).description() << bufferGroupInfos.at(y).axes().at(x).units() << bufferGroupInfos.at(y).axes().at(x).size() << bufferGroupInfos.at(y).axes().at(x).isUniform() << bufferGroupInfos.at(y).axes().at(x).start() << bufferGroupInfos.at(y).axes().at(x).increment();
-				}
-			}
-		}
+	AMDSClientTCPSocket * clientTCPSocket = new AMDSClientTCPSocket(hostLineEdit->text(), portLineEdit->text().toInt());
```

Comment:
yes, they are:
 - if invalid message type, delete socket
- is socket error, delete socket
- is message data received (not continuous), delete socket